### PR TITLE
Improve error detection & handling when invoking SVN

### PR DESCRIPTION
**Pull request by bright-tools**
As discussed on #241, try and be a bit more robust when executing the SVN command.

@dregad, I tried to keep my implementation close to the work that you did in the hope it would ease the review process.  Please take a look & let me know if you want any changes prior to merge.  I've done some basic testing.

### CODE REVIEW

File: SourceSVN/SourceSVN.php [hunk 1]
```diff
@@ -317,12 +308,46 @@ public function import_latest( $p_repo ) {
 		return $this->import_full( $p_repo );
 	}
 
-	private function svn_check() {
-		$svn = self::svn_call();
+	/**
+	 * Execute SVN command, catching & raising errors in both
+	 *   execution and output
+	 * @param string Command and any parameters
+	 * @param SourceRepo Repository to access
```
Comment (dregad) [inline SourceSVN/SourceSVN.php]: Add the param names to the PHPDoc block, i.e.
```
@param string $p_cmd Command and any parameters
```

File: SourceSVN/lang/strings_english.txt [hunk 2]
```diff
@@ -24,5 +24,5 @@ $s_plugin_SourceSVN_svnssl = 'SVN: Trust All SSL Certs<br/><span class="small">(
 $s_plugin_SourceSVN_winstart = 'SVN: Use Windows `start`<br/><span class="small">(Requires a configured path to binary)</span>';
 
 $s_plugin_SourceSVN_error_path_invalid = 'Path to Subversion binary is invalid, inaccessible or not a directory.';
-$s_plugin_SourceSVN_error_run_svn = 'shell_exec() failed to execute Subversion.';
-
+$s_plugin_SourceSVN_error_svn_run = 'Failed to execute Subversion.  Command returned "%1$s"';
+$s_plugin_SourceSVN_error_svn_cmd = 'svn execution error: "%1$s".';
```
Comment (dregad) [inline SourceSVN/lang/strings_english.txt]: The error messages are very similar - is it really worth having 2 different strings ?

**Comment by bright-tools** [comment]
@dregad , thanks for the review!

I hopefully fixed the PHPDoc block

With regard to the errors, I've tried to refine them (the redundant parameter on one of the probably did not help).  Intent is that the first error is triggered in the case that `proc_open()` fails (i.e. not possible to execute SVN), while the other error is triggered if the SVN process returns to stderr.  It seemed like a distinction worth making, but let me know what you think.

File: SourceSVN/SourceSVN.php [hunk 3]
```diff
@@ -317,12 +308,46 @@ public function import_latest( $p_repo ) {
 		return $this->import_full( $p_repo );
 	}
 
-	private function svn_check() {
-		$svn = self::svn_call();
+	/**
+	 * Execute SVN command, catching & raising errors in both
+	 *   execution and output
+	 * @param string p_cmd Command and any parameters
+	 * @param SourceRepo p_repo Repository to access
```
Comment (dregad) [inline SourceSVN/SourceSVN.php]: You missed the `$` in front of the param name :wink:

**Comment by dregad** [comment]
WRT the error messages, I think its good following your latest commit.

**Comment by bright-tools** [comment]
Argh!  You can tell how much I've used PHPDoc mark-up, can't you? :grin:

**Comment by dregad** [comment]
@bright-tools Thanks for your contribution !

As a side note and if you don't mind, I would prefer when you submit pull requests, that you *rebase* your branch on top of upstream master, instead of merging it (that's in the interest of keeping a clean git history). Thanks !

**Comment by bright-tools** [comment]
Noted, will do!

**Pull request merged**